holoviews/util/__init__.py

```diff
@@ -215,6 +237,39 @@ def _options_error(cls, opt, objtype, backend, valid_options):
                              'across all extensions. No similar options '
                              'found.' % (opt, objtype))
 
+    @classmethod
+    def _completer_reprs(cls, options, namespace=None, ns=None):
+        """
+        Given a list of Option objects (such as those returned from
+        OptsSpec.parse_options) or an %opts or %%opts magic string,
+        return a list of corresponding completer reprs. The namespace is
+        typically given as 'hv' if fully qualified namespaces are
+        desired.
```

Comment:
Not sure I follow the bit about the namespace, also having ``namespace`` and ``ns`` is confusing me here.

Reply:
Agreed. Here `ns` is the namespace as an actual namespace dictionary, e.g containing things like `Cycle` or `dim` that should be accepted by the parser (if a string is provided). And `namespace` is the string as the prefix in the repr e.g if you don't want to import `opts` from `holoviews` but imported holoviews as `hv` then you can use `namespace='hv'` so the reprs are things like `'hv.opts.Curve(...)'` that can be executed as reprs.

The problem is that even after explaining it I can't think of better names! Maybe `namespace` could be `namespace_prefix` or `prefix`?

Reply:
I'd vote for `ns` to be `ns_dict` or `namespace_dict`, and `namespace` to be `namespace_str` or `ns_str`.  But I'm not sure how these objects will be used, so don't take my word for it...